docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,206 @@
+
+#### NEW: Bootstrap Token Secrets
+
+Bootstrap tokens are stored and managed via Kubernetes secrets in the `kube-system` namespace.  They have type `bootstrap.kubernetes.io/token`.
+
+The following keys are on the secret data:
+* **token-id**. As defined above.
+* **token-secret**. As defined above.
+* **expiration**. After this time the token should be automatically deleted.
+* **usage-bootstrap-signing**. Set to `true` to indicate this token should be used for signing bootstrap configs.  If omitted or some other string, it defaults to `false`.
+
+These secrets can be named anything but it is suggested that they be named `bootstrap-token-<token-id>`.
+
+**QUESTION:** Should we also spec out now how we can use this token for TLS bootstrap.
+
+#### Quick Primer on JWS
+
+[JSON Web Signatures](https://tools.ietf.org/html/rfc7515) are a way to sign, serialize and verify a payload.  It supports both symmetric keys (aka shared secrets) along with asymmetric keys (aka public key infrastructure or key pairs).  The JWS is split in to 3 parts:
+1. a header about how it is signed
+2. the clear text payload
+3. the signature.
+
+There are a couple of different ways of encoding this data -- either as a JSON object or as a set of BASE64URL strings for including in headers or URL parameters.  In this case, we are using a shared secret and the HMAC-SHA256 signing algorithm and encoding it as a JSON object.  The popular JWT (JSON Web Tokens) specification is a type of JWS.
+
+The JWS specification [describes how to encode](https://tools.ietf.org/html/rfc7515#appendix-F) "detached content".  In this way the signature is calculated as normal but the content isn't included in the signature.
+
+#### NEW: `kube-public` namespace
+
+Kubernetes ConfigMaps are per-namespace and are generally only visible to principals that have read access on that namespace.  To create a config map that *everyone* can see, we introduce a new `kube-public` namespace.  This namespace, by convention, is readable by all users (including those not authenticated).
+
+#### NEW: `cluster-info` ConfigMap
+
+A new well known ConfigMap will be created in the `kube-public` namespace called `cluster-info`.
+
+Users configuring the cluster (and eventually the cluster itself) will update the `kubeconfig` key here with the limited `kubeconfig` above.
+
+A new controller is introduced that will watch for both new/modified bootstrap tokens and changes to the `cluster-info` ConfigMap.  As things change it will generate new JWS signatures. These will be saved under ConfigMap keys of the pattern `jws-kubeconfig-<token-id>`.
+
+In addition, `jws-kubeconfig-<token-id>-hash` will be set to the MD5 hash of the contents of the `kubeconfig` data.  This will be in the form of `md5:d3b07384d113edec49eaa6238ad5ff00`.  This is done so that the controller can detect which signatures need to be updated without reading all of the tokens.
+
+This controller will also delete tokens that are past their expiration time.
```

Comment:
This would be an internal component.  It would be logically layered on top of core and could easily be broken out into another component in the future.  Keeping it in the controller manager or the api server would let us use unified watch caches and reduce the impact of having another system.  When talking to @smarterclayton he was supportive of this for those concerns.